reference/conanfile/tools/google/bazeldeps.rst

```diff
@@ -0,0 +1,327 @@
+
+Since Conan 1.62.0, you can put all the files generated by ``BazelDeps`` into another folder using the ``bazel_layout``:
+
+.. code-block:: python
+    :caption: conanfile.py
+
+    from conan import ConanFile
+    from conan.tools.google import BazelDeps, bazel_layout
+
+    class App(ConanFile):
+        settings = "os", "arch", "compiler", "build_type"
+        requires = "zlib/1.2.11"
+
+        def layout(self):
+            # DEPRECATED: Default generators folder will be "conan" in Conan 2.x
+            self.folders.generators = "conan"
+            bazel_layout(self)
+
+        def generate(self):
+            bz = BazelDeps(self)
+            bz.generate()
+
+
+Running again the :command:`conan install` command, we now get this structure:
+
+.. code-block:: bash
+
+    $ conan install .
+    .
+    ├── conan
+    │   ├── BUILD.bazel
+    │   ├── dependencies.bzl
+    │   └── zlib
+    │       └── BUILD.bazel
+    └── conanfile.py
+
+
+Now your Conan-bazel files were generated in the *conan/* folder, so your WORKSPACE will look like:
+
+.. code-block:: python
+    :caption: WORKSPACE
+
+    load("@//conan:dependencies.bzl", "load_conan_dependencies")
+    load_conan_dependencies()
+
+
+
+Build requirements
+++++++++++++++++++
+
+Available since: `1.62.0 <https://github.com/conan-io/conan/releases/tag/1.62.0>`_
+
+When you have a **build-requirement**, by default, the Bazel files are not generated.
+But you can activate it using the **build_context_activated** attribute:
+
+.. code-block:: python
+
+    def build_requirements(self):
+        self.tool_requires("my_tool/0.0.1")
+
+    def layout(self):
+        # DEPRECATED: Default generators folder will be "conan" in Conan 2.x
+        self.folders.generators = "conan"
+        bazel_layout(self)
+
+    def generate(self):
+        bz = BazelDeps(self)
+        # generate the build-mytool/BUILD.bazel file for the tool require
+        bz.build_context_activated = ["my_tool"]
+        bz.generate()
+
+
+Running the :command:`conan install` command, the structure created is as follows:
+
+.. code-block:: bash
+
+    $ conan install . -pr:b default
+    .
+    ├── conan
+    │   ├── BUILD.bazel
+    │   ├── build-my_tool
+    │   │   └── BUILD.bazel
+    │   └── dependencies.bzl
+    └── conanfile.py
+
+Notice that *my_tool* Bazel folder is prefixed with ``build-`` which indicates that it's being used in the build context.
+
+
+Properties
+++++++++++
+
+Available since: `1.62.0 <https://github.com/conan-io/conan/releases/tag/1.62.0>`_
+
+The following properties affect the ``BazelDeps`` generator:
+
+- **bazel_target_name** property will define the name of the target declared in the *<REPOSITORY>/BUILD.bazel*
+- **bazel_repository_name** property will define the name of the folder where
+
+These properties can be defined at global ``cpp_info`` level or at component level (``bazel_repository_name`` is only
+available for the global level).
```

Comment:
Maybe move this to the description of the properties?

Reply:
It's ok for me too